pennylane_lightning/src/algorithms/AdjointDiff.hpp

```diff
@@ -546,12 +664,13 @@ template <class T = double> class AdjointJacobian {
      * @param ops_matrices Matrix definition of an operation if unsupported.
      * @return const OpsData<T>
      */
-    const OpsData<T> createOpsData(
+    auto createOpsData(
         const std::vector<std::string> &ops_name,
```

Comment:
Wouldn't pass-by-value and using `std::move` be a better practice here?

Reply:
Potentially, though `OpsData` has 2 constructors (one for ref, one for copy and move). In this instance it may be better, though clang did not recognize the need for modification here.

It could be how the method is constructed. Though, this method is only used in C++ tests, as the bound method for Python is exposed differently.